### DISCUSSION TITLE
Adding differential gene expression to the to the SwitchList

### DISCUSSION THREAD

**Discussion by jdhahbi** [kvittingseerup/IsoformSwitchAnalyzeR#119] [General]
Is there an example on how to add differential expression from edgeR or DESeq2 results to the SwitchList?
You recommend to use match() to overwrite 'exampleSwitchList$isoformFeatures$gene_q_value and exampleSwitchList$isoformFeatures$iso_q_value' with the corresponding FDR corrected P-values of a differential expression analysis. 
I would appreciate it very much if you show me how to perform this step. My students and I are really enjoying this great package. 
Thank you in advance
Joseph Dhahbi

**Comment by kvittingseerup** [comment]
Unfortunately, there is not. The problem is that depending on how you run DESeq2 the conditions might be encoded differently making it hard to do a once-size-fits-all solution. The code below can be used as inspiration but will probably have to be adapted to your data.

If you only have 1 condition the pseudo code will look something like this:

```
switchAnalyzeRlist$$isoformFeatures$gene_q_value <- deseq2res$padj[ match(
    switchAnalyzeRlist$$isoformFeatures$gene_id, deseq2res$gene_id
)]
```

If you have more than 1 comparison you will have to manually add them as columns in the DESeq2 results table and then ensure the match function also uses that information. The code would looks something like this:

```
switchAnalyzeRlist$$isoformFeatures$gene_q_value <- deseq2res$padj[ match(
    str_c(
         switchAnalyzeRlist$$isoformFeatures$gene_id,
         switchAnalyzeRlist$$isoformFeatures$condition_1,
         switchAnalyzeRlist$$isoformFeatures$condition_2
    ),
    str_c(
         deseq2res$gene_id,
         deseq2res$condition_1,
         deseq2res$condition_2
    )
)]
```

Hope this helps.

Cheers
Kristoffer

**Reply by jdhahbi** [reply]
It was useful. Thank you so much.
